crates/ruff_python_ast/src/traversal.rs

```diff
@@ -0,0 +1,108 @@
+//! Utilities for manually traversing a Python AST.
+use rustpython_ast::{ExceptHandler, Stmt, Suite};
+use rustpython_parser::ast;
+
+/// Given a [`Stmt`] and its parent, return the [`Suite`] that contains the [`Stmt`].
+pub fn suite<'a>(stmt: &'a Stmt, parent: &'a Stmt) -> Option<&'a Suite> {
```

Comment:
Nit: `containing_suite`? 

This is one of the places where a `SuiteStmt` would help a lot because the parent would then be the suite directly. 

Would it make sense to not only search for the statement, but also its index to avoid two passes?